### ISSUE TITLE
integrate xyz

### ISSUE THREAD

**Issue by davidchambers** [ramda/ramda-fantasy#19]
Integrating [xyz][1] is a prerequisite for #15.

[1]: https://github.com/davidchambers/xyz

**Comment by TheLudd** [comment]
Isn't that what `npm version` is for?
https://docs.npmjs.com/cli/version

**Comment by davidchambers** [comment]
They're solving the same problem. xyz is more flexible and allows any number of prepublish scripts to be run as preconditions and/or to modify files for inclusion in the tagged commit.

Ramda, for example, specifies a script argument in its [makefile][1]:

```make
XYZ = node_modules/.bin/xyz --repo git@github.com:ramda/ramda.git --script scripts/prepublish
```

The [prepublish][2] script updates version numbers in __README.md__ and rebuilds __dist/ramda.js__ and __dist/ramda.min.js__:

```bash
#!/usr/bin/env bash
set -e

README="$(cat README.md)"
README="${README//"$PREVIOUS_VERSION/ramda.min.js"/"$VERSION/ramda.min.js"}"
README="${README//"${PREVIOUS_VERSION%.*}/ramda.min.js"/"${VERSION%.*}/ramda.min.js"}"
echo "$README" >README.md
git add README.md

rm -f dist/ramda{,.min}.js
make dist/ramda{,.min}.js
git add dist/ramda{,.min}.js
```

We've been using xyz to publish Ramda for several months, and it works well. I assume this project will adopt the same approach, though it needn't.


[1]: https://github.com/ramda/ramda/blob/v0.10.0/Makefile#L9
[2]: https://github.com/ramda/ramda/blob/v0.10.0/scripts/prepublish

**Comment by buzzdecafe** [comment]
let's do it--worth keeping tooling as consistent as possible across these repos i think. and we have had fewer screwups since xyz was introduced

**Comment by buzzdecafe** [comment]
closed by #42
